napari/utils/events/_tests/test_evented_model.py

```diff
@@ -238,6 +239,64 @@ class User(EventedModel):
     assert user1_events.call_count == 0
 
 
+def test_update_with_inner_model_union():
+    class Inner(EventedModel):
+        w: str
+
+    class AltInner(EventedModel):
+        x: str
+
+    class Outer(EventedModel):
+        y: int
+        z: Union[Inner, AltInner]
+
+    original = Outer(y=1, z=Inner(w='a'))
+    updated = Outer(y=2, z=AltInner(x='b'))
+
+    original.update(updated, recurse=False)
+
+    assert original == updated
```

Comment:
maybe i'm not thinking straight at all here, but i would have expected this to work only if recurse=True. Can you add another test with the `recurse=True` with these same classes that checks the expected result, it would just help me understand 100% what was happening here. thanks!!!

Reply:
Note that `recurse=True` is the default/existing case, so some of the other tests cover that case already.

In this test, omitting `recurse=False` or replacing it with `recurse=True` will raise an exception because `update` will effectively try to do `original.z.x = 'b'` and will fail because the existing instance of `Inner` has no field named `x` (only `z`).

I feel like adding a test to cover this exceptional case probably isn't worth maintaining. Instead I added some more to the docstring for the recurse parameter. What do you think?

Reply:
Ok great - I understand why `recurse=True` will fail now. I was then tripped up on why the updated original would equal updated without recursion - i was thinking, without recursion the lower layers won't be updated - but they don't actually end up mattering as updating the top layer just overwrites everything. this all makes sense to me now